benchexec/tools/btormc.py

```diff
@@ -0,0 +1,47 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec.result as result
+import benchexec.tools.template
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    Tool info for BtorMC -- A Bounded Model Checker for Btor2
+    """
+
+    REQUIRED_PATHS = ["build/"]
```

Comment:
I followed the instructions in Boolector's [README](https://github.com/Boolector/boolector?tab=readme-ov-file#linux-and-unix-like-os).
The compiled statically-linked binary of BtorMC is `build/bin/btormc`.
We could of course specify other directory to place the build artifacts.

> everything outside of it being irrelevant for execution

This is indeed the case for BtorMC.
Since the executable is statically-linked, only the binary itself is required.
So I think we could remove the `REQUIRED_PATHS` since no other file is needed.

Reply:
Addressed in 7427ddf

Reply:
I think following this we should also remove `build/` from the executable search path. And maybe if there is really just the executable and nothing else, also remove `bin/`? If I take the tool and copy it somewhere, I would not expect to be forced to create a `bin/` subdirectory just in order to satisfy the tool-info module.

Reply:
If I understand `ToolLocator.find_executable` correctly,
it searches for executable under `./` and `./build/bin/` ([lines](https://github.com/sosy-lab/benchexec/blob/e22ca4443024a58dd7ea5828755acf833b028c53/benchexec/tools/template.py#L371-L374)).

If a user (like me), follows the build instructions to compile the tool,
he/she does not need to specify additional parameters to locate the executable when using BenchExec.

Reply:
Ok, now I just notice that when `--tool-directory` is specified,
`ToolLocator` only looks into `tool_dir/subdir/`.

Reply:
`subdir` removed in baadd81